### PR TITLE
Spike/es lint basic

### PR DESCRIPTION
**Pull request by PatFawbertMills**
Introducing a minimalist ESLint ruleset into the front-end code.

A lot of the changes are related to dependancies to useEffect, some of which I have asked esLint to ignore for functional reasons. I aim to circle back round to these at a later date to reevaluate if it needs a refactor.

Important:
The search and the landing page (index) has had a refactor also. We were overusing component state rather than deriving it from our searchCriteria object, for instance the `pageNumber`.

### CODE REVIEW

**Comment by PatFawbertMills** [comment]
ESLint checks will now run as part of the CI just FYI

File: frontend/src/components/EmbeddedPDF.tsx [hunk 1]
```diff
@@ -11,18 +11,21 @@ type TProps = {
...
+  const pdfPreview = usePDFPreview(document);
+  // Ensure the instance of the PDF client is not reset on render 
+  // otherwise we lose the ability to interact with the pdf
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const { createPDFClient, passageIndexChangeHandler } = useMemo(() => pdfPreview, [document]);
 
   useEffect(() => {
     if (containerRef?.current) {
-      createPDFClient(passageIndex);
+      createPDFClient();
```
Comment (diversemix) [inline frontend/src/components/EmbeddedPDF.tsx:22]: nice improvement :+1:

File: frontend/src/pages/search/index.tsx [hunk 2]
```diff
@@ -231,25 +227,11 @@ const Search = () => {
   }, [hits]);
 
   useDidUpdateEffect(() => {
-    setOffset(searchCriteria?.offset);
-    setCurrentCategoryIndex();
+    window.scrollTo(0, 0);
+    // setOffset(searchCriteria?.offset);
```
Comment (diversemix) [inline frontend/src/pages/search/index.tsx:231]: I don't feel confident to review this change from cold.

**Review by diversemix — APPROVED** [review]
lgtm - some nice improvements a long the way - and a comment on a section I don'f feel confident in reviewing :+1: Nice work as always!

File: frontend/src/pages/search/index.tsx (hunk 2 above)
Comment (PatFawbertMills) [reply frontend/src/pages/search/index.tsx]: Sorry didn't mean to leave this comment in. I've been trying to remove the reliance on setting component state in multiple places.

**Review by diversemix — APPROVED** [review]
Looks great

**Pull request merged**